contrib/hyperscan/matching/input_matchers/source/config.cc

```diff
@@ -24,8 +24,48 @@ Config::createInputMatcherFactoryCb(const Protobuf::Message& config,
 #ifdef HYPERSCAN_DISABLED
   throw EnvoyException("X86_64 architecture is required for Hyperscan.");
 #else
+  // Hyperscan's API requires vectors of expressions, flags and IDs for matching database
+  // compilation.
   return [hyperscan_config, &factory_context]() {
-    return std::make_unique<Matcher>(hyperscan_config, factory_context.threadLocal());
+    int size = hyperscan_config.regexes().size();
+    std::vector<const char*> expressions;
+    std::vector<unsigned int> flags;
+    std::vector<unsigned int> ids;
+    expressions.reserve(size);
+    flags.reserve(size);
+    ids.reserve(size);
+    for (const auto& regex : hyperscan_config.regexes()) {
+      expressions.push_back(regex.regex().c_str());
+      unsigned int flag = 0;
+      if (regex.caseless()) {
+        flag |= HS_FLAG_CASELESS;
+      }
+      if (regex.dot_all()) {
+        flag |= HS_FLAG_DOTALL;
+      }
+      if (regex.multiline()) {
+        flag |= HS_FLAG_MULTILINE;
+      }
+      if (regex.allow_empty()) {
+        flag |= HS_FLAG_ALLOWEMPTY;
+      }
+      if (regex.utf8()) {
```

Comment:
should we make the default behavior match RE2 more closely by flipping the defaults for some of these?

Or is that too late because you already checked in a proto-file?

Reply:
I do not want to change behavior of Hyperscan input matcher. I will leave this choice to user in Hyperscan input matcher, and make the default behavior match RE2 closely in Hyperscan regex engine.

Reply:
Yes. In Hyperscan regex engine, I will set all these flags to match RE2's behavior.